pkg/controller/disruption/disruption.go

```diff
@@ -301,6 +305,29 @@ func (dc *DisruptionController) getScaleController(controllerRef *metav1.OwnerRe
 	return &controllerAndScale{scale.UID, scale.Spec.Replicas}, nil
 }
 
+func verifyGroupKind(controllerRef *metav1.OwnerReference, expectedKind string, expectedGroups []string) (bool, error) {
+	gv, err := schema.ParseGroupVersion(controllerRef.APIVersion)
+	if err != nil {
+		return false, err
+	}
+
+	groupMatch := false
+	for _, group := range expectedGroups {
+		if group == gv.Group {
+			groupMatch = true
+			break
+		}
+	}
+	if !groupMatch {
+		return false, nil
+	}
+
+	if controllerRef.Kind == expectedKind {
```

Comment:
nit: move this first and short-circuit the parse and group search and return `false, nil` early if this doesn't match

Reply:
Yeah, this looks cleaner. I have updated the function.